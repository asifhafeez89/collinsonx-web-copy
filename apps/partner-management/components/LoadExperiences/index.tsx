import InputSelect from '@collinsonx/design-system/components/inputselect/index';
import { Experience } from '@collinsonx/utils';
import { useQuery } from '@collinsonx/utils/apollo';
import { setItem } from '@collinsonx/utils/lib/index';
import { getSearchExperiences } from '@collinsonx/utils/queries';
import { SELECTED_LOUNGE } from 'config';

export interface LoadExperiencesProps {
  selectedExperience?: Experience;
  onExperienceSelected: (experience: Experience) => void;
}

const LoadExperiences = ({
  selectedExperience,
  onExperienceSelected,
}: LoadExperiencesProps) => {
  const { loading, error, data, refetch } = useQuery<{
    searchExperiences: Experience[];
  }>(getSearchExperiences);

  const experiencesFiltered = data?.searchExperiences.map(
    (experience: Experience) => {
      return {
        value: experience.id,
        label: `${experience.loungeName}${
          experience.location?.terminal
            ? ' - ' + experience.location?.terminal
            : ''
        }`,
      };
    }
  );

  return (
    <div>
      {data?.searchExperiences && (
        <InputSelect
          styles={{
            root: {
              width: '400px',
            },
          }}
          data={experiencesFiltered ?? []}
          onChange={async (id: string) => {
            const newExperience = data?.searchExperiences.filter(
              (item: Experience) => item.id === id
            )[0]! as Experience;

            onExperienceSelected(newExperience);

            setItem(SELECTED_LOUNGE, JSON.stringify(newExperience));
          }}
          value={selectedExperience?.id}
        />
      )}
    </div>
  );
};

export default LoadExperiences;
