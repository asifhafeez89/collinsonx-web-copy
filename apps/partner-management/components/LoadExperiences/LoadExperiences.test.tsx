import LoadExperiences from './index';
import '@testing-library/jest-dom/extend-expect';
import {
  fireEvent,
  render,
  screen,
  waitFor,
  getNodeText,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';

jest.mock('@collinsonx/utils/lib/index', () => {
  return jest.fn(() => ({
    setItem: () => {
      return [];
    },
  }));
});

jest.mock('@collinsonx/utils/queries', () => {
  return jest.fn(() => ({
    getSearchExperiences: () => {
      return { searchExperiences: [] };
    },
  }));
});

describe('<LoadExperiences />', () => {
  it('should render', () => {
    const experience = {
      id: '1ccc3807-a7ed-5a3a-ada8-fd37ac1ab941',
      loungeName: 'Clubrooms Birmingham - Additional Fee Applies',
      bookings: [],
      invitations: [],
      partners: [],
    };

    const dropdown = render(
      <LoadExperiences
        selectedExperience={experience}
        onExperienceSelected={() => console.log()}
      />
    );

    expect(dropdown).toMatchSnapshot();
  });

  it('should render  element', async () => {
    const experience = {
      id: '1ccc3807-a7ed-5a3a-ada8-fd37ac1ab941',
      loungeName: 'Clubrooms Birmingham - Additional Fee Applies',
      bookings: [],
      invitations: [],
      partners: [],
    };

    const dropdown = render(
      <LoadExperiences
        selectedExperience={experience}
        onExperienceSelected={() => console.log()}
      />
    );

    expect(
      dropdown.getByDisplayValue(
        'Clubrooms Birmingham - Additional Fee Applies'
      )
    ).toBeInTheDocument();
  });
});
