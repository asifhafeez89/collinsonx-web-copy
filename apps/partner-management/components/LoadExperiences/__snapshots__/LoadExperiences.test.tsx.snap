// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadExperiences /> should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": .emotion-2 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  line-height: 1.55;
  width: 400px;
  color: #000!important;
}

.emotion-2 label {
  color: #000000;
  font-weight: 400;
  margin-bottom: 8px;
}

.emotion-2 input[data-invalid] {
  border-color: red;
}

.emotion-5 {
  position: relative;
}

.emotion-5:has(input:disabled) .emotion-10 {
  display: none;
}

.emotion-8 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  height: 2.25rem;
  -webkit-tap-highlight-color: transparent;
  line-height: calc(2.25rem - 0.125rem);
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  resize: none;
  box-sizing: border-box;
  font-size: 0.875rem;
  width: 100%;
  color: #000;
  display: block;
  text-align: left;
  border: 0.0625rem solid #ced4da;
  background-color: #fff;
  -webkit-transition: border-color 100ms ease;
  transition: border-color 100ms ease;
  min-height: 2.25rem;
  padding-left: calc(2.25rem  / 3);
  padding-right: 2.25rem;
  border-radius: 0.25rem;
}

.emotion-8:focus,
.emotion-8:focus-within {
  outline: none;
  border-color: #228be6;
}

.emotion-8:disabled,
.emotion-8[data-disabled] {
  background-color: #f1f3f5;
  color: #909296;
  opacity: 0.6;
  cursor: not-allowed;
  pointer-events: none;
}

.emotion-8:disabled::-webkit-input-placeholder {
  color: #909296;
}

.emotion-8:disabled::-moz-placeholder {
  color: #909296;
}

.emotion-8:disabled:-ms-input-placeholder {
  color: #909296;
}

.emotion-8[data-disabled]::-webkit-input-placeholder {
  color: #909296;
}

.emotion-8[data-disabled]::-moz-placeholder {
  color: #909296;
}

.emotion-8[data-disabled]:-ms-input-placeholder {
  color: #909296;
}

.emotion-8:disabled::placeholder,
.emotion-8[data-disabled]::placeholder {
  color: #909296;
}

.emotion-8[data-invalid] {
  color: #fa5252;
  border-color: #fa5252;
}

.emotion-8[data-invalid]::-webkit-input-placeholder {
  opacity: 1;
  color: #fa5252;
}

.emotion-8[data-invalid]::-moz-placeholder {
  opacity: 1;
  color: #fa5252;
}

.emotion-8[data-invalid]:-ms-input-placeholder {
  opacity: 1;
  color: #fa5252;
}

.emotion-8[data-invalid]::placeholder {
  opacity: 1;
  color: #fa5252;
}

.emotion-8[data-with-icon] {
  padding-left: 2.25rem;
}

.emotion-8::-webkit-input-placeholder {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #adb5bd;
  opacity: 1;
}

.emotion-8::-moz-placeholder {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #adb5bd;
  opacity: 1;
}

.emotion-8:-ms-input-placeholder {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #adb5bd;
  opacity: 1;
}

.emotion-8::placeholder {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #adb5bd;
  opacity: 1;
}

.emotion-8::-webkit-inner-spin-button,
.emotion-8::-webkit-outer-spin-button,
.emotion-8::-webkit-search-decoration,
.emotion-8::-webkit-search-cancel-button,
.emotion-8::-webkit-search-results-button,
.emotion-8::-webkit-search-results-decoration {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
}

.emotion-8[type=number] {
  -moz-appearance: textfield;
}

.emotion-8:not(:disabled) {
  cursor: pointer;
}

.emotion-8:not(:disabled)::selection {
  background-color: transparent;
}

.emotion-9 {
  position: absolute;
  top: 0;
  bottom: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 2.25rem;
}

.mantine-1surk1h:has(input:disabled) .emotion-10 {
  display: none;
}

.mantine-1e6nbp3:has(input:disabled) .emotion-10 {
  display: none;
}

<body>
    <div>
      <div>
        <div
          class="emotion-0 emotion-1 emotion-2"
        >
          <div
            aria-controls="load-experiences--selected"
            aria-expanded="false"
            aria-haspopup="listbox"
            class=""
            role="combobox"
            tabindex="-1"
          >
            <input
              type="hidden"
              value="1ccc3807-a7ed-5a3a-ada8-fd37ac1ab941"
            />
            <div
              class="emotion-3 emotion-4 emotion-5"
            >
              <input
                aria-autocomplete="list"
                aria-invalid="false"
                autocomplete="off"
                class="emotion-6 emotion-7 emotion-8"
                data-mantine-stop-propagation="false"
                id="load-experiences--selected"
                readonly=""
                role="dialog"
                type="search"
                value="Clubrooms Birmingham - Additional Fee Applies"
              />
              <div
                class="emotion-9 emotion-10 emotion-11"
              >
                <span />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .emotion-2 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  line-height: 1.55;
  width: 400px;
  color: #000!important;
}

.emotion-2 label {
  color: #000000;
  font-weight: 400;
  margin-bottom: 8px;
}

.emotion-2 input[data-invalid] {
  border-color: red;
}

.emotion-5 {
  position: relative;
}

.emotion-5:has(input:disabled) .emotion-10 {
  display: none;
}

.emotion-8 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  height: 2.25rem;
  -webkit-tap-highlight-color: transparent;
  line-height: calc(2.25rem - 0.125rem);
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  resize: none;
  box-sizing: border-box;
  font-size: 0.875rem;
  width: 100%;
  color: #000;
  display: block;
  text-align: left;
  border: 0.0625rem solid #ced4da;
  background-color: #fff;
  -webkit-transition: border-color 100ms ease;
  transition: border-color 100ms ease;
  min-height: 2.25rem;
  padding-left: calc(2.25rem  / 3);
  padding-right: 2.25rem;
  border-radius: 0.25rem;
}

.emotion-8:focus,
.emotion-8:focus-within {
  outline: none;
  border-color: #228be6;
}

.emotion-8:disabled,
.emotion-8[data-disabled] {
  background-color: #f1f3f5;
  color: #909296;
  opacity: 0.6;
  cursor: not-allowed;
  pointer-events: none;
}

.emotion-8:disabled::-webkit-input-placeholder {
  color: #909296;
}

.emotion-8:disabled::-moz-placeholder {
  color: #909296;
}

.emotion-8:disabled:-ms-input-placeholder {
  color: #909296;
}

.emotion-8[data-disabled]::-webkit-input-placeholder {
  color: #909296;
}

.emotion-8[data-disabled]::-moz-placeholder {
  color: #909296;
}

.emotion-8[data-disabled]:-ms-input-placeholder {
  color: #909296;
}

.emotion-8:disabled::placeholder,
.emotion-8[data-disabled]::placeholder {
  color: #909296;
}

.emotion-8[data-invalid] {
  color: #fa5252;
  border-color: #fa5252;
}

.emotion-8[data-invalid]::-webkit-input-placeholder {
  opacity: 1;
  color: #fa5252;
}

.emotion-8[data-invalid]::-moz-placeholder {
  opacity: 1;
  color: #fa5252;
}

.emotion-8[data-invalid]:-ms-input-placeholder {
  opacity: 1;
  color: #fa5252;
}

.emotion-8[data-invalid]::placeholder {
  opacity: 1;
  color: #fa5252;
}

.emotion-8[data-with-icon] {
  padding-left: 2.25rem;
}

.emotion-8::-webkit-input-placeholder {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #adb5bd;
  opacity: 1;
}

.emotion-8::-moz-placeholder {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #adb5bd;
  opacity: 1;
}

.emotion-8:-ms-input-placeholder {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #adb5bd;
  opacity: 1;
}

.emotion-8::placeholder {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #adb5bd;
  opacity: 1;
}

.emotion-8::-webkit-inner-spin-button,
.emotion-8::-webkit-outer-spin-button,
.emotion-8::-webkit-search-decoration,
.emotion-8::-webkit-search-cancel-button,
.emotion-8::-webkit-search-results-button,
.emotion-8::-webkit-search-results-decoration {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
}

.emotion-8[type=number] {
  -moz-appearance: textfield;
}

.emotion-8:not(:disabled) {
  cursor: pointer;
}

.emotion-8:not(:disabled)::selection {
  background-color: transparent;
}

.emotion-9 {
  position: absolute;
  top: 0;
  bottom: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 2.25rem;
}

.mantine-1surk1h:has(input:disabled) .emotion-10 {
  display: none;
}

.mantine-1e6nbp3:has(input:disabled) .emotion-10 {
  display: none;
}

<div>
    <div>
      <div
        class="emotion-0 emotion-1 emotion-2"
      >
        <div
          aria-controls="load-experiences--selected"
          aria-expanded="false"
          aria-haspopup="listbox"
          class=""
          role="combobox"
          tabindex="-1"
        >
          <input
            type="hidden"
            value="1ccc3807-a7ed-5a3a-ada8-fd37ac1ab941"
          />
          <div
            class="emotion-3 emotion-4 emotion-5"
          >
            <input
              aria-autocomplete="list"
              aria-invalid="false"
              autocomplete="off"
              class="emotion-6 emotion-7 emotion-8"
              data-mantine-stop-propagation="false"
              id="load-experiences--selected"
              readonly=""
              role="dialog"
              type="search"
              value="Clubrooms Birmingham - Additional Fee Applies"
            />
            <div
              class="emotion-9 emotion-10 emotion-11"
            >
              <span />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
